### PR TITLE
Adding global options to specify strong option in validate block

### PR DESCRIPTION
**Pull request by yayohei**
I want use strong parameter whole my project because permitted_params is needless to cast every variables. so i wanna add `strong: true` option to validates block.
```ruby
class RecipesController < ApplicationController
  validates :create, strong: true do
    integer :user_id
    integer :category_id
  end

  def create
    respond_with Recipe.create(permitted_params)
  end
end
```
_I wonder what is the best naming of test case instead of global strong._

### CODE REVIEW

**Comment by taiki45** [comment]
>because permitter_params is needless to cast every variables

What is your requirements? Why `permitted_params` (I guess you got a typo) is needless?

For the naming, I think it's not "global" but is "per validation block" or something.

File: lib/weak_parameters/validator.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 module WeakParameters
   class Validator
-    attr_reader :block, :controller
+    attr_reader :block, :controller, :options
```
Comment (taiki45) [inline lib/weak_parameters/validator.rb]: Why do we need this?

File: spec/dummy/app/controllers/concerns/with_global_strong.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      object :strong_body do
+        list :items, :object, description: 'some items' do
+          string :name
+          integer :strong_price, strong: true
+        end
+      end
+    end
+  end
+
+  def create
```
Comment (taiki45) [inline spec/dummy/app/controllers/concerns/with_global_strong.rb]: Don't use mix-in to unify business logics. Just write this two times at each controller in this case.

File: spec/requests/global_strong_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+require "spec_helper"
```
Comment (taiki45) [inline spec/requests/global_strong_spec.rb]: I think this non-API controller test is not valuable, so can we drop this test and the related controller?

**Review by taiki45 — CHANGES_REQUESTED** [review]
Adding this "per validation block option" sounds good to me.

The current implementation accepts and passes all options: https://github.com/r7kamura/weak_parameters#available-options. I think accepting `only` option or something is confusing for users. So we need add some validation logics to prevent this.

In addition, we need a documentation for this.

**Comment by yayohei** [comment]
Thank you for quick response 😄 I'm sorry for typo and poor English. 
My purpose is I don't want to add to strong option to the every parameters. But maybe my request not match to original purpose of this gem.
```ruby
  #  many strong option appeared...
  validates :create do
    integer :user_id, strong: true
    integer :category_id, strong: true
  end

  validates :update do
    integer :user_id, strong: true
    integer :category_id, strong: true
  end
```
I will fix your request 🙇

File: lib/weak_parameters/validator.rb (hunk 1 above)
Comment (yayohei) [reply lib/weak_parameters/validator.rb]: OK, removed.

File: spec/dummy/app/controllers/concerns/with_global_strong.rb (hunk 2 above)
Comment (yayohei) [reply spec/dummy/app/controllers/concerns/with_global_strong.rb]: Fixed.
https://github.com/r7kamura/weak_parameters/pull/35/commits/b84a2f28ab6a3f05551bf5dafb48f351b0f5d2f9#diff-f8cd801f019ef57590adfe5ad17a34a8R5

File: spec/requests/global_strong_spec.rb (hunk 3 above)
Comment (yayohei) [reply spec/requests/global_strong_spec.rb]: Removed this spec.

**Comment by yayohei** [comment]
Hello, @taiki45. I updated PR.
Naming is a very difficult. How about this? not so long, and extendable syntax. :sweat_smile:
```ruby
  validates :create, block_option: { strong: true } do
    integer :user_id, strong: true
    integer :category_id, strong: true
  end
```

**Pull request closed without merging**